myhpom/tests/test_state_requirement_model.py

```diff
@@ -0,0 +1,77 @@
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+from django.db import transaction, IntegrityError
+from myhpom.models import State, StateRequirement
+
+
+class StateRequirementModelTestCase(TestCase):
+    def setUp(self):
+        # get a state that has both global and non-global requirements: NC
+        self.state = State.objects.get(name="NC")
+        self.state_reqs = list(StateRequirement.objects.filter(state=self.state))
+        self.global_reqs = list(StateRequirement.objects.filter(state=None))
+
+    def test__create_invalid(self):
+        """
+        cannot create StateRequirement without the required fields: position, text.
+        """
+        invalid_data = [
+            {'text': ''},  # needs position, valid text
+            {'text': 'a non-existent global requirement'},  # needs position
+            {'position': len(self.global_reqs)},  # needs valid text
+            {'position': len(self.global_reqs), 'text': ''},  # needs valid text
+        ]
+        for data in invalid_data:
+            with transaction.atomic():
+                self.assertRaises(ValidationError, StateRequirement.objects.create, **data)
+
+    def test__create_not_unique(self):
+        """
+        cannot create StateRequirement that violates uniqueness: (state, order), (state, text)
+        """
+        invalid_data = [
+            # state-specific
+            {
+                'state': self.state,
+                'position': self.state_reqs[0].position,  # violate uniqueness for state, position
+                'text': 'a non-existent state requirement',
+            },
+            {
+                'state': self.state,
+                'position': len(self.state_reqs),
+                'text': self.state_reqs[0].text,  # violate uniqueness for state, text
+            },
+            # global -- WON'T WORK WITHOUT DATABASE TRIGGER
+            # -- (state=null, position) and (state=null, text) can be non-unique
+            # {
+            #     'state': None,
+            #     'position': self.global_reqs[0].position,  # violate uniqueness for state, position
+            #     'text': 'a non-existent global requirement',
+            # },
+            # {
+            #     'state': None,
+            #     'position': len(self.global_reqs),
+            #     'text': self.global_reqs[0].text,  # violate uniqueness for state, text
+            # },
```

Comment:
Does this commented out block need to be another test case?

Reply:
Yes, I just needed to spend a little more time and raise the IntegrityError in the model.